gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/StructuredQuery.java

```diff
@@ -714,32 +687,34 @@ B addGroupBy(String property, String... others) {
       return self();
     }
 
-    B mergeFrom(DatastoreV1.Query queryPb) {
+    B mergeFrom(com.google.datastore.v1beta3.Query queryPb) {
       if (queryPb.getKindCount() > 0) {
         kind(queryPb.getKind(0).getName());
       }
-      if (queryPb.hasStartCursor()) {
+      if (!queryPb.getStartCursor().isEmpty()) {
         startCursor(new Cursor(queryPb.getStartCursor()));
       }
-      if (queryPb.hasEndCursor()) {
+      if (!queryPb.getEndCursor().isEmpty()) {
         endCursor(new Cursor(queryPb.getEndCursor()));
       }
-      if (queryPb.hasOffset()) {
-        offset(queryPb.getOffset());
-      }
+      offset(queryPb.getOffset());
       if (queryPb.hasLimit()) {
-        limit(queryPb.getLimit());
+        limit(queryPb.getLimit().getValue());
       }
       if (queryPb.hasFilter()) {
-        filter(Filter.fromPb(queryPb.getFilter()));
+        Filter currFilter = Filter.fromPb(queryPb.getFilter());
+        if (currFilter != null) {
+          filter(currFilter);
+        }
       }
-      for (DatastoreV1.PropertyOrder orderByPb : queryPb.getOrderList()) {
+      for (com.google.datastore.v1beta3.PropertyOrder orderByPb : queryPb.getOrderList()) {
         addOrderBy(OrderBy.fromPb(orderByPb));
       }
-      for (DatastoreV1.PropertyExpression projectionPb : queryPb.getProjectionList()) {
+      for (com.google.datastore.v1beta3.Projection projectionPb
+           : queryPb.getProjectionList()) {
         addProjection(Projection.fromPb(projectionPb));
       }
-      for (DatastoreV1.PropertyReference groupByPb : queryPb.getGroupByList()) {
+      for (com.google.datastore.v1beta3.PropertyReference groupByPb : queryPb.getDistinctOnList()) {
```

Comment:
That is indeed different.

So, I assume you suggest we should rename it in our API and call it distinct.

One option is to make distinct an attribute of a Projection field.
e.g.
```java
Query.projectionEntityQueryBuilder()
    .kind(kind)
    .projection(Projection.distinct("age"), Projection.property("name"))
```
But then it is not as convenient to support DISTINCT on all properties.

Another option would be the replace the groupBy option with distinct
and either have a special token for all (e.g. "*") or provide it via an empty distinct call.

e.g.
```java
Query.projectionEntityQueryBuilder()
    .kind(kind)
    .projection(Projection.property("age"), Projection.property("name"))
    .distinct("age")

Query.projectionEntityQueryBuilder()
    .kind(kind)
    .projection(Projection.property("age"), Projection.property("name"))
    .distinct()
```

What do you think? BTW, if we go with the latter I think we can make project be simple
string rather than a `Projection` class, like so:
```java
Query.projectionEntityQueryBuilder()
    .kind(kind)
    .projection("age", "name")
    .distinct("age")
```